drivers: modem: gsm: allow the user to get modem information

bbilas opened the pull request:
Make modem_info structure public in order to allow the user
to get modem information using gsm_ppp_modem_info function.

Signed-off-by: Bartosz Bilas <b.bilas@grinn-global.com>

bbilas committed:
drivers: modem: gsm: allow the user to get modem information

Make modem_info structure public in order to allow the user
to get modem information using gsm_ppp_modem_info function.
Move modem info query commands into separate function
that's called only once during gsm configuring because
there is no necessity to re-querying modem since
they should not change.

Signed-off-by: Bartosz Bilas <b.bilas@grinn-global.com>

jukkar reviewed (commented):
> I'd prefer that the modem info is only queried once instead of during every `gsm_ppp_start`

I agree, the information should not change so getting it once is enough.
Overall, this looks reasonable.

bbilas committed:
drivers: modem: gsm: move header file into modem directory

There is a specified location for modem drivers so move this
header right there.

Signed-off-by: Bartosz Bilas <b.bilas@grinn-global.com>

bbilas committed:
drivers: modem: gsm: extend modem info struct about RSSI value

Add RSSI member into gsm_ppp_modem_info structure in order to
get that information using the gsm_ppp_modem_info function.

Signed-off-by: Bartosz Bilas <b.bilas@grinn-global.com>

jukkar reviewed (commented):
Looks good. Minor nit about the commit subjects, I suggest to change them to be similar looking to something like
`drivers: modem: gsm: ......`
no need to add the ppp there, it is just useless information here.
